### ISSUE TITLE
Not parse Optional[Decimal]

### ISSUE THREAD

**Issue by heckad** [lidatong/dataclasses-json#137]
```
from dataclasses import dataclass
from decimal import Decimal
from typing import Optional

from dataclasses_json import DataClassJsonMixin


@dataclass
class Price(DataClassJsonMixin):
    price: Optional[Decimal]


def test_simple():
    p = Price(price=Decimal("1.2"))
    np = Price.from_json(p.to_json())
    assert p == np
```

**Comment by heckad** [comment]
Will it be fixed?

**Comment by tstehr** [comment]
I had the same problem, worked around by adding a custom decoder:

```
@dataclass
class Price(DataClassJsonMixin):
    price: Optional[Decimal] = field(
        metadata=config(decoder=lambda val: Decimal(val) if val is not None else None)
    )
```

**Comment by lidatong** [comment]
Hi -- thanks for raising this. It should be addressed in latest release (v0.3.3).
